src/bin/bindgen.rs

```diff
@@ -217,6 +217,16 @@ pub fn main() {
 
     let mut bind_args: Vec<_> = env::args().collect();
 
+    let clang_version = clang_version();
+
+    if cfg!(feature = "llvm_stable") {
+        assert_eq!(clang_version, "3.8.0");
+    } else {
+        assert_eq!(clang_version, "3.9.0");
```

Comment:
I think if we want to parse it we should be conservative. That means not panicking if the reported version is not the one we expect (that means having `clang_version` return an `Option`, either `Option<String>`, `Option<(u32, u32)>` (major, minor), `Option<Version>` (being version a new type), and not use `expect` but early return everywhere in that function).

IMO we should:

 * Always log the string we get in `extract_clang_version`.
 * If we check a version, never check patch numbers.
 * `error!` (maybe panic is ok if there's any way of disable it via a flag or anything) if we found a version that doesn't match what we expect.